core/src/rpc_subscriptions.rs

```diff
@@ -1,44 +1,58 @@
 //! The `pubsub` module implements a threaded subscription service on client RPC request
 
+use crate::bank_forks::BankForks;
 use bs58;
 use core::hash::Hash;
 use jsonrpc_core::futures::Future;
 use jsonrpc_pubsub::typed::Sink;
 use jsonrpc_pubsub::SubscriptionId;
+use serde::Serialize;
 use solana_runtime::bank::Bank;
 use solana_sdk::account::Account;
 use solana_sdk::pubkey::Pubkey;
 use solana_sdk::signature::Signature;
 use solana_sdk::transaction;
+use solana_vote_api::vote_state::MAX_LOCKOUT_HISTORY;
 use std::collections::HashMap;
-use std::sync::RwLock;
+use std::sync::{Arc, RwLock};
 
-type RpcAccountSubscriptions = RwLock<HashMap<Pubkey, HashMap<SubscriptionId, Sink<Account>>>>;
+pub type Confirmations = usize;
+
+type RpcAccountSubscriptions =
+    RwLock<HashMap<Pubkey, HashMap<SubscriptionId, (Sink<Account>, Confirmations)>>>;
 type RpcProgramSubscriptions =
-    RwLock<HashMap<Pubkey, HashMap<SubscriptionId, Sink<(String, Account)>>>>;
-type RpcSignatureSubscriptions =
-    RwLock<HashMap<Signature, HashMap<SubscriptionId, Sink<Option<transaction::Result<()>>>>>>;
+    RwLock<HashMap<Pubkey, HashMap<SubscriptionId, (Sink<(String, Account)>, Confirmations)>>>;
+type RpcSignatureSubscriptions = RwLock<
+    HashMap<Signature, HashMap<SubscriptionId, (Sink<transaction::Result<()>>, Confirmations)>>,
+>;
 
 fn add_subscription<K, S>(
-    subscriptions: &mut HashMap<K, HashMap<SubscriptionId, Sink<S>>>,
+    subscriptions: &mut HashMap<K, HashMap<SubscriptionId, (Sink<S>, Confirmations)>>,
     hashmap_key: &K,
+    confirmations: Option<Confirmations>,
     sub_id: &SubscriptionId,
     sink: &Sink<S>,
 ) where
     K: Eq + Hash + Clone + Copy,
     S: Clone,
 {
+    let confirmations = confirmations.unwrap_or(0);
```

Comment:
Hmm, I sorta want to make the default MAX_LOCKOUT_HISTORY instead of 0.   This gives first-time users the most consistency when they develop their dapps.  If they want to unbuckle the seat beat to get faster responses and can accept the consistency trade off then they can by adding the confirmations option.  

This is probably worthy of a separate issue though to first better understand the impact of this on existing users

Reply:
Yeah, I considered that myself, but finally opted to go with a default closest to current behavior.
One approach I was thinking of playing around with is to establish 2 subscriptions for an interesting event: one that notifies right away, and one that notifies later (possibly at MAX_LOCKOUT_HISTORY) as a "confidence check". The viability of that certainly depends on the number of interesting events a dapp is trying to follow.